projects/npm-tools/packages/npm-scripts/contrib/prettier/prettier.js

```diff
@@ -992,6 +992,33 @@ module.exports = {
 	version: '2.1.2',
 };
 
+function getWorkspaceRoot(filepath) {
+	let current = filepath;
+
+	while (true) {
+		const parent = path.dirname(current);
+
+		const candidate = path.join(parent, 'settings.gradle');
+
+		if (fs.existsSync(candidate)) {
```

Comment:
We will need to do more than just check if settings.gradle exists because this file exists at `liferay-portal/modules/settings.gradle` as well. I think you also need to read the content of the file and check for the workspace plugin

```suggestion
		if (fs.existsSync(candidate) && fs.readFileSync(candidate).includes('apply plugin: "com.liferay.workspace"')) {
```